### DISCUSSION TITLE
Query filters for entities who have Handle<A> as a component

### DISCUSSION THREAD

**Discussion by stinkytoe** [bevyengine/bevy#16043] [Ideas]
Has there ever been any thought into query filters which yield for handles as components, when the assets behind those handles are loaded, changed, removed, etc.. ?

I noticed that handles are de facto components, and can be added to entities via commands.spawn(..) or similar.

~~Looking at the systems for managing Sprite, I saw that (at least for sprites) the technique is to respond to the AssetEvent messages, and then crawl through a query to find the associated entities which contain that handle (in this case as members of the Sprite struct).~~
Edit: my mistake, the handle is part of SpriteBundle, not Sprite.

Would there be value in adding QueryFilter variants such as AssetLoaded<A>, AssetModified<A>, etc.. which would allow the user to yield on these events directly in the query syntax and not need to search on their own through the AssetEvent messages?

Just an interesting quality of life improvement that I was thinking of. Please let me know if this seems like a good idea, or if it's a terrible idea for reasons I'm not foreseeing.

**Comment by torsteingrindvik** [comment]
https://github.com/bevyengine/bevy/issues/15716

FYI `Handle<_>` will no longer be a component.

**Reply by stinkytoe** [reply]
Interesting. I see the reasoning for that. So any new crates/plugins should go with the 'handle as a field of a component' pattern is what I am gleaning from this.

Just to clarify, the intended workflow is:
* Spawn a component with a `Handle<A>` as a field, like `MyComponent { handle: Handle<A> }`
* Use a system with `EventReader<AssetEvent<A>>` and `Query<(&MyComponent, ...)>` parameters which will:
  - Trigger on `Added {..}`, `Modified {..}`, etc events as appropriate
  - Search through the query for a component with a matching handle id
  - Do the appropriate actions (add components/children/etc..)

It would still be nice to have some kind of sugar to simplify this, but that can probably wait until the Bevy interface is more solidified. This solution works for me in the interim.

Thank you for the help!
